app/mutations/application/deleteAttachment.ts

```diff
@@ -0,0 +1,33 @@
+import { graphql } from "react-relay";
+import RelayModernEnvironment from "relay-runtime/lib/store/RelayModernEnvironment";
+import {
+  deleteAttachmentMutationVariables,
+  deleteAttachmentMutation as deleteAttachmentMutationType,
+} from "deleteAttachmentMutation.graphql";
+import BaseMutation from "mutations/BaseMutation";
+
+const mutation = graphql`
+  mutation deleteAttachmentMutation(
```

Comment:
**To be done in a separate PR:**
We will want to handle delete differently for previous versions (hard delete if attachment is from current version, soft delete it attachment was added in a previous version).
We can either define two mutations here and conditionally pass them to the performMutation() call below, or we can override the behaviour with a custom SQL delete mutation.